File: java/src/main/java/com/workshop/architecture/fitness/inside/port/inbound/ActivateMembershipResult.java
package com.workshop.architecture.fitness.inside.port.inbound;

import java.time.LocalDate;

public record ActivateMembershipResult(
        String membershipId,
        String customerId,
        String planId,
        int planPrice,
        int planDuration,
        String status,
        LocalDate startDate,
        LocalDate endDate,
        String invoiceId,
        String externalInvoiceId,
        LocalDate invoiceDueDate
) {
}

File: python/src/workshop_api/fitness/clean_architecture/use_case/port/inbound/activate_membership_result.py
from dataclasses import dataclass
from datetime import date


@dataclass(frozen=True)
class ActivateMembershipResult:
    membership_id: str
    customer_id: str
    plan_id: str
    plan_price: int
    plan_duration: int
    status: str
    start_date: date
    end_date: date
    invoice_id: str
    external_invoice_id: str
    invoice_due_date: date
